### ISSUE TITLE
In the course of actual training, we need to display the corresponding ACC, loss, val_acc, and val_loss curves to observe the effect of the model training, but I didn't find this method in deepchem.

### ISSUE THREAD

**Issue by zhouhao-learning** [deepchem/deepchem#1240]
In the process of actual training, we need to display the corresponding ACC, loss, val_acc, val_loss curve, in order to observe the effect of the model training, but in deepchem, I have not found this method of.Is there a callback function that can. Ard only display loss curves, and does the validation set do, validation?

**Comment by lilleswing** [comment]
Calculating these statistics is currently on the user.
If you turn on logging we do display the training loss periodically.

Calculating validation performance and validation loss require an additional validation dataset and take time that would otherwise be spent training the model, so it is currently on the user to train for the number of epochs they wish before doing validation..

Writing tools to for this would be a great addition to the framework though!  PR welcome.

**Comment by vsag96** [comment]
@lilleswing  @rbharath How do you want the functionality to be, should it be a new class with a method which takes in arguments? I am comfortable with python and pytorch, also don't think writing code for accuracy is that hard be it any library/deep learning, I want to take a stab at this, unsure of where to start. Can you please guide me on what is it that you expect so I can start working on it.

**Comment by rbharath** [comment]
@vsag96 Thanks for volunteering!

To start with, I'd suggest a new class in `contrib/`. The contribution barrier is a lot lower. Once the API is stabilized, can be migrated into the main project in a future PR.

Want to start a pull request and mark it [WIP]? We can provide feedback on the work in progress.

**Comment by vsag** [comment]
Sure I'll read tensorflow and it's workflow a bit and I'll submit a PR maybe by Sunday so bear with me thanks!

**Comment by vsag96** [comment]
Due to some work related emergencies I won't be able to submit a pr, If anyone else is interested please start working on it.extremely sorry for the inconvenience.

**Comment by rbharath** [comment]
Closing this old issue
